### ISSUE TITLE
`isinstance(arg, dict)` branch does fully capture `TypedDict` argument

### ISSUE THREAD

**Issue by randolf-scholz** [astral-sh/ty#3237]
### Summary

I expected no errors in the example below, since the `isinstance(spec, dict)` branch should capture any `TypedDict`. It seems `ty` does not simplify `(Config[T@initialize] & ~Top[dict[Unknown, Unknown]])` to `Never`.

### `main.py`

```py
from typing import TypedDict

class Config[T](TypedDict): pass

def initialize[T](spec: T | type[T] | Config[T], /) -> T:
    if isinstance(spec, type):
        raise NotImplementedError
    if isinstance(spec, dict):
        raise NotImplementedError
    return spec
```

```
Return type does not match returned value: expected `T@initialize`, found `(T@initialize & ~type & ~Top[dict[Unknown, Unknown]]) | (Config[T@initialize] & ~Top[dict[Unknown, Unknown]])` (invalid-return-type) [Ln 10, Col 12]
```

### `ty.json`

```json
{
    "environment": {
        "python-version": "3.14"
    },
    "rules": {
        "undefined-reveal": "ignore"
    }
}
```

**Comment by AlexWaygood** [comment]
Thanks! This is https://github.com/astral-sh/ty/issues/1130, and will shortly be fixed by https://github.com/astral-sh/ruff/pull/23946
